blockchain/blockindex.go

```diff
@@ -161,12 +185,35 @@ func (node *blockNode) Ancestor(height int32) *blockNode {
 		return nil
 	}
 
-	n := node
-	for ; n != nil && n.height != height; n = n.parent {
-		// Intentionally left blank
+	indexWalk := node
+	heightWalk := node.height
+
+	for heightWalk > height {
+		heightSkip := getSkipHeight(heightWalk)
+		heightSkipPrev := getSkipHeight(heightWalk - 1)
+
+		if indexWalk.ancestor != nil &&
```

Comment:
Quite a mouth-full of a conditional... 

It also isn't immediately clear the significance (nor origin) of the `-2` here.

Reply:
Got rid of everything here and changed the logic to be simpler.

idk why that -2 is there as well... Some sipa magic